scipy/optimize/tests/test_zeros.py

```diff
@@ -753,3 +754,28 @@ def f_p(x):
         zeros.newton(f, 1.0, f_p)
     root = zeros.newton(f, complex(10.0, 10.0), f_p)
     assert_allclose(root, complex(0.0, 1.0))
+
+
+def test_newton_backtracks_if_out_of_domain():
+    """Test that newton does not get stuck if one function evaluation returns NaN"""
+    # See https://github.com/scipy/scipy/issues/12625
+
+    def f(x):
+        return np.log(x)
+
+    assert_warns(RuntimeWarning, zeros.newton, f, 15.0, disp=False)
+    with pytest.raises(
```

Comment:
Testing two warnings, or one warning and one error is... weird, I'm not sure if this is the best way to do it

Reply:
this looks fine to me